### PR TITLE
reporter: add GitHub support

### PR DESCRIPTION
**Pull request by GiedriusS**
Add GitHub reporter support. All of the problems in the summary are
submitted as (multi-)line comments on a given pull request. Add some
small tests to try out this functionality. I have also done some ad-hoc
tests.

Also, exit with 1 when there are problems with severity Bug or higher.
This is so that it would be possible to run `pint ci` in Jenkins so that
it would report problems on GitHub & exit with 1 to indicate that a step
has failed. All of the users which depend on the exit code being 0 can
simply ignore the errors with `||:` in Bash.

cc @prymitive

### CODE REVIEW

File: cmd/pint/ci.go [hunk 1]
```diff
@@ -70,13 +72,53 @@ func actionCI(c *cli.Context) (err error) {
...
 	if len(bySeverity) > 0 {
 		log.Info().Fields(bySeverity).Msg("Problems found")
 	}
 
-	return submitReports(reps, summary)
+	if err := submitReports(reps, summary); err != nil {
+		return fmt.Errorf("submitting reports: %w", err)
+	}
+
+	return cli.Exit("", exitCode)
```
Comment (prymitive) [inline cmd/pint/ci.go:122]: I think I would prefer to retain current behaviour and simply return an error, like `return fmt.Errorf("Found at least one issue")`.
Error returned here will cause exit with non-zero code - https://github.com/cloudflare/pint/blob/main/cmd/pint/main.go#L72-L76

Comment (GiedriusS) [reply cmd/pint/ci.go]: Thanks! I have implemented this in https://github.com/cloudflare/pint/pull/18/commits/44d4c07fd233cb7cb758d1be1ceaee575ab47fe8. I retained the same message as in `cmd/pint/lint.go`: https://github.com/cloudflare/pint/blob/main/cmd/pint/lint.go#L62

I hope that's OK.

**Review by prymitive — APPROVED** [review]

**Comment by prymitive** [comment]
Looks like go.mod/sum needs to be updated:

```
 #23 9.481 go: github.com/cloudflare/pint/internal/reporter: package golang.org/x/oauth2 imported from implicitly required module; to add missing requirements, run:
#23 9.481 	go get golang.org/x/oauth2@v0.0.0-20210514164344-f6687ab2804c
```

**Comment by GiedriusS** [comment]
> Looks like go.mod/sum needs to be updated:
> 
> ```
>  #23 9.481 go: github.com/cloudflare/pint/internal/reporter: package golang.org/x/oauth2 imported from implicitly required module; to add missing requirements, run:
> #23 9.481 	go get golang.org/x/oauth2@v0.0.0-20210514164344-f6687ab2804c
> ```

Should be fixed by https://github.com/cloudflare/pint/pull/18/commits/e7b0e6702f8509d09ef67faf4870774dc7622f4d.

**Comment by prymitive** [comment]
Thanks!

**Pull request merged**